src/openloco/windowmgr.cpp

```diff
@@ -1406,4 +1416,20 @@ namespace openloco::ui::windowmgr
         regs.bp = dY;
         call(0x004C6B09, regs); // openrct2: viewport_redraw_after_shift
     }
+
+    static void sub_4CF456()
+    {
+        close(window_type::dropdown, 0);
+
+        for (ui::window* w = _windows_end - 1; w >= _windows; w--)
+        {
+            if ((w->flags & window_flags::stick_to_back) != 0)
+                continue;
+
+            if ((w->flags & window_flags::stick_to_front) != 0)
+                continue;
+
+            close(w);
```

Comment:
I believe the loop should start over after closing a window.

Reply:
I'm not sure how to implement this. Help would be appreciated

Reply:
(Also, this appears to be `window_close_all` (in openrct), but it's done in a different way there.

Reply:
How about just `w = _windows_end;`? It'll `w--` when the iteration continues.